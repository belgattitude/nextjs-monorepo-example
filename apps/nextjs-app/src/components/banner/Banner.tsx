import { Speaker as SpeakerIcon, Close as XIcon } from '@mui/icons-material';
import type { FC } from 'react';

type Props = {
  children?: never;
};

export const Banner: FC<Props> = () => {
  return (
    <div className="bg-indigo-600">
      <div className="mx-auto max-w-7xl p-3 sm:px-6 lg:px-8">
        <div className="flex flex-wrap items-center justify-between">
          <div className="flex w-0 flex-1 items-center">
            <span className="flex rounded-lg bg-indigo-800 p-2">
              <SpeakerIcon className="h-6 w-6 text-white" aria-hidden="true" />
            </span>
            <p className="ml-3 truncate font-medium text-white">
              <span className="md:hidden">We announced a new product!</span>
              <span className="hidden md:inline">
                Big news! We're excited to announce a brand new product.
              </span>
            </p>
          </div>
          <div className="order-3 mt-2 w-full shrink-0 sm:order-2 sm:mt-0 sm:w-auto">
            <a
              href="@/components/banner/Banner#"
              className="flex items-center justify-center rounded-md border border-transparent bg-white py-2 px-4 text-sm font-medium text-indigo-600 shadow-sm hover:bg-indigo-50"
            >
              Learn more
            </a>
          </div>
          <div className="order-2 shrink-0 sm:order-3 sm:ml-3">
            <button
              type="button"
              className="-mr-1 flex rounded-md p-2 hover:bg-indigo-500 focus:outline-none focus:ring-2 focus:ring-white sm:-mr-2"
            >
              <span className="sr-only">Dismiss</span>
              <XIcon className="h-6 w-6 text-white" aria-hidden="true" />
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};
