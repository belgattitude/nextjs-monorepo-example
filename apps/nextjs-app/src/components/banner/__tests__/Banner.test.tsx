import renderer from 'react-test-renderer';
import { Banner } from '../Banner';

it('should match snapshot', () => {
  const tree = renderer.create(<Banner />).toJSON();
  expect(tree).toMatchSnapshot();
});
