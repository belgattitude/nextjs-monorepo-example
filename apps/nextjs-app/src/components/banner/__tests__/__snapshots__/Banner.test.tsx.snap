// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<div
  className="bg-indigo-600"
>
  <div
    className="mx-auto max-w-7xl p-3 sm:px-6 lg:px-8"
  >
    <div
      className="flex flex-wrap items-center justify-between"
    >
      <div
        className="flex w-0 flex-1 items-center"
      >
        <span
          className="flex rounded-lg bg-indigo-800 p-2"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium h-6 w-6 text-white css-i4bv87-MuiSvgIcon-root"
            data-testid="SpeakerIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 2H7c-1.1 0-2 .9-2 2v16c0 1.1.9 1.99 2 1.99L17 22c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-5 2c1.1 0 2 .9 2 2s-.9 2-2 2c-1.11 0-2-.9-2-2s.89-2 2-2zm0 16c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
        </span>
        <p
          className="ml-3 truncate font-medium text-white"
        >
          <span
            className="md:hidden"
          >
            We announced a new product!
          </span>
          <span
            className="hidden md:inline"
          >
            Big news! We're excited to announce a brand new product.
          </span>
        </p>
      </div>
      <div
        className="order-3 mt-2 w-full shrink-0 sm:order-2 sm:mt-0 sm:w-auto"
      >
        <a
          className="flex items-center justify-center rounded-md border border-transparent bg-white px-4 py-2 text-sm font-medium text-indigo-600 shadow-sm hover:bg-indigo-50"
          href="@/components/banner/Banner#"
        >
          Learn more
        </a>
      </div>
      <div
        className="order-2 shrink-0 sm:order-3 sm:ml-3"
      >
        <button
          className="-mr-1 flex rounded-md p-2 hover:bg-indigo-500 focus:outline-none focus:ring-2 focus:ring-white sm:-mr-2"
          type="button"
        >
          <span
            className="sr-only"
          >
            Dismiss
          </span>
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium h-6 w-6 text-white css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
